clang/lib/Sema/SemaExpr.cpp

```diff
@@ -62,7 +62,34 @@
 
 using namespace clang;
 using namespace sema;
-
+class EnumNameVisitor : public RecursiveASTVisitor<EnumNameVisitor> {
+private:
+  std::string EnumName;
+public:
+  void VisitDeclRefExpr(DeclRefExpr *Node) {
+    const Decl *D = Node->getDecl();
+    // Check if the declaration referenced is an enum constant
+    if (const auto *ECD = dyn_cast<EnumConstantDecl>(D)) {
+      if (const auto *ED = dyn_cast<EnumDecl>(ECD->getDeclContext())) {
+        // Construct the fully qualified name
+        EnumName = ED->getName().str() + "::" + ECD->getName().str();
```

Comment:
You can still store `EnumName` as a `llvm::StringRef` since its easy to construct from either `std::string` or `const char*`:
```
        llvm::StringRef EnumName;
        EnumName = ED->getName().str() + "::" + ECD->getName().str(); // refer StringRef (const std::string &Str)
```

Reply:
`llvm::StringRef` can only store a single value. For example:

```cpp
llvm::StringRef EnumName;
EnumName = ED->getName();
```

This assignment does not produce an error. However, `llvm::StringRef` does not support concatenation of multiple values, which results in an error when I try to perform concatenation directly:

```
error: no viable overloaded '='
   75 |         EnumName = ED->getName() + "::" + ECD->getName();
```